### ISSUE TITLE
Support multiple teams

### ISSUE THREAD

**Issue by davidstanke** [GoogleCloudPlatform/fourkeys#281]
Many users want to use a single installation of Four Keys to track metrics across multiple teams. At present, it's possible to submit webhooks from multiple repos (or multiple teams' repos) to the same endpoint. And then you could modify the grafana config to support filtering on that.

But this is currently an exercise left to the reader... let's make it built-in.

**Comment by r0mdau** [comment]
In Gitlab there is a concept of groups, subgroups and git repository, relations are 1..n 1..n. 

It is possible to set up a webhook at the group level and all events comming from subgroups git repositories will be sent to the fourkeys unique webhook.

End goal : It could be nice to have a dropdown in Grafana variables allowing to select a group and then it filters and permit to select a subgroup. To allow a multi view for KPIs at different levels.

**Comment by madeline-scyphers** [comment]
I'm about to do this for work, so I can implement this. 

To get at repo filtering, you need access to the metadata field. I was thinking of adding it to the views so the views don't need to be joined back to `event_raw` in the dashboard. Saving that join could help performance. But I am open to leaving the views as is and joining the views back to `events_raw` on the dashboard side if people think that would be better.

**Comment by madeline-scyphers** [comment]
We could also add to the new table `events_enriched`, something that normalizes repo name for all supported event types and and the parsers that use it (and groups and subgroups to if the sources supports it), that way instead of every event type having to do it differently in the view or in the dashboard query, they would all be the same. Thoughts on that idea?

**Comment by madeline-scyphers** [comment]
Change the new `events` view to this
```
# events table
SELECT raw.id,
       CASE 
         WHEN source LIKE 'github%' OR (source LIKE 'gitlab%' AND REGEXP_CONTAINS(LOWER(metadata), r'repository')) 
           THEN JSON_EXTRACT_SCALAR(metadata, '$.repository.name')
         WHEN source LIKE 'gitlab%' THEN JSON_EXTRACT_SCALAR(metadata, '$.project.name')
       END AS repo_name,
       raw.event_type,
       raw.time_created,
       raw.metadata,
       enr.enriched_metadata,
       raw.signature,
       raw.msg_id,
       raw.source
FROM four_keys.events_raw raw
LEFT JOIN four_keys.events_enriched enr
    ON raw.signature = enr.events_raw_signature
```

And then change the `deployments`, `changes`, and `incidents` views to be based on this view instead of `events_raw` and you don't have to change any of the parsers. Let those views pass repo_name through for filtering in the dashboard. Also we could still pass metadata along just in case people want to filter by things like groups or other things

**Comment by martasd** [comment]
Hi @madeline-scyphers. We have multiple repos in Github for which we would like to track metrics. How far did you get with this idea and its implementation? Do you have draft PR for this addition? I would appreciate anything you have to share regarding this. Thanks!

**Comment by madeline-scyphers** [comment]
@martasd I don't have a draft PR. I was waiting on feedback before I got started. And now I am no longer assigned to this project through work, so I won't be completing it. Sorry!

**Comment by martasd** [comment]
FYI: Creating the new `events` view as described above worked for me. 👍
